### ISSUE TITLE
Can't run VM based test from March 2017 or so

### ISSUE THREAD

**Issue by obadz** [NixOS/nixpkgs#30262]
I'm on 2b99882ae6 (17.09) and trying to `git bisect` a bug that was introduced between 17.03 and 17.09. I can't run any VM based tests older than some past commit on the master branch.

For instance, when checking out 22da5d0246 and trying to run this test, I get the error:

```
$ nix-build nixos/tests/ecryptfs.nix 
these derivations will be built:
  /nix/store/g71m5a7byh8p8fzhvmn79afhbab5raf2-vm-test-run-ecryptfs.drv
building path(s) ‘/nix/store/as93gawp675hxf9ch46b45njp2ccpk7f-vm-test-run-ecryptfs’
starting VDE switch for network 1
vde_switch: Could not set the VDE ctl directory '/tmp/nix-build-vm-test-run-ecryptfs.drv-0/vde1.ctl' permissions: Operation not permitted
vde_switch: Could not remove ctl socket '/tmp/nix-build-vm-test-run-ecryptfs.drv-0/vde1.ctl': No such file or directory
cannot start vde_switch at /nix/store/0y4sxpn8wbgd6xqn3m8jb1mssjn5m7g2-nixos-test-driver/bin/.nixos-test-driver-wrapped line 42.
cleaning up
builder for ‘/nix/store/g71m5a7byh8p8fzhvmn79afhbab5raf2-vm-test-run-ecryptfs.drv’ failed with exit code 255
error: build of ‘/nix/store/g71m5a7byh8p8fzhvmn79afhbab5raf2-vm-test-run-ecryptfs.drv’ failed
```
Any one knows why? Is this a qemu version issue? Is there a work around or do I need to cherry-pick qemu updates?

**Comment by edolstra** [comment]
This is probably caused by Nix no longer permitting the creation of setuid/setgid files. The fix is 0996ea8a68096b596b27678dfdd0f4acde7b92de.

**Comment by obadz** [comment]
That was it, thank you.
